pkg/controller/node/nodecontroller.go

```diff
@@ -575,15 +600,29 @@ func (nc *NodeController) monitorNodeStatus() error {
 		}
 	}
 	nc.handleDisruption(zoneToNodeConditions, nodes)
+	nc.updateEvictionMetric(Evictions10Minutes, nc.evictions10Minutes)
+	nc.updateEvictionMetric(Evictions1Hour, nc.evictions1Hour)
 
 	return nil
 }
 
+func (nc *NodeController) updateEvictionMetric(metric *prometheus.GaugeVec, data *evictionData) {
+	data.slideWindow()
+	zones := data.getZones()
+	metric.Reset()
```

Comment:
IIUC, the purpose of it is to remove zones that are no longer there, right? If so, is there a method on the metric to remove a given value for a given label?